### PR TITLE
[MI-2754] Add input panel

### PR DESCRIPTION
**Pull request by ayusht2810**
image

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

File: src/containers/HomePage/inputPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import {Button, Input} from '@fluentui/react-northstar';
+
+const InputPanel = () => {
+    const [teamsChannelUrl, setTeamsChannelUrl] = useState('');
+    const [mattermostChannelUrl, setMattermostChannelUrl] = useState('');
+
+    // Error states
+    const [teamsChannelUrlError, setTeamsChannelUrlError] = useState(false);
+    const [mattermostChannelUrlError, setMattermostChannelUrlError] = useState(false);
```
Comment (avas27JTG) [inline src/containers/HomePage/inputPanel.tsx]:
```suggestion
   type FormConfig {
      placeholder: string;
      label: string;
      value: string;
      error?: string | boolean;
      isRequired?: boolean;
   }
   
    enum ConnectFormFields {
     teamsChannelUrl = 'teamsChannelUrl'
     mattermostChannelURL = 'mattermostChannelURL'
    }
    
    const connectFormData: Record<ConnectFormFields, FormConfig> = {
    [ConnectFormFields.teamsChannelURL]: {
      ...
      }
    }
    
   const [connectForm, setConnectForm] = useState<Record<ConnectFormFields, Form>>({...connectFormData});
```
create a common form data structure

File: src/containers/HomePage/inputPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    };
+
+    const handleMattermostChannelUrlChange = (e: React.SyntheticEvent) => {
+        setMattermostChannelUrlError(false);
+        setMattermostChannelUrl((e.target as HTMLInputElement).value);
+    };
+
+    const handleChannelConnect = () => {
+        if (!teamsChannelUrl) {
+            setTeamsChannelUrlError(true);
```
Comment (avas27JTG) [inline src/containers/HomePage/inputPanel.tsx]:
```suggestion
            setConnectForm({...connectForm, teamsChannelUrl: {...connectForm.teamsChannelUrl, error: true}});
```

File: src/containers/HomePage/inputPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        setTeamsChannelUrl((e.target as HTMLInputElement).value);
+    };
+
+    const handleMattermostChannelUrlChange = (e: React.SyntheticEvent) => {
+        setMattermostChannelUrlError(false);
+        setMattermostChannelUrl((e.target as HTMLInputElement).value);
+    };
+
+    const handleChannelConnect = () => {
+        if (!teamsChannelUrl) {
```
Comment (avas27JTG) [inline src/containers/HomePage/inputPanel.tsx]:
```suggestion
        if (!connectForm.teamsChannelUrl.value) {
```

File: src/types/index.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export enum ConnectFormFields {
```
Comment (avas27JTG) [inline src/types/index.ts]: prevent using `enum` create objects instead see https://dev.to/ivanzm123/dont-use-enums-in-typescript-they-are-very-dangerous-57bh

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**